resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -384,14 +390,16 @@ private[spark] class KubernetesClusterSchedulerBackend(scheduler: TaskSchedulerI
     }
   }
 
-  private val executorCleanupRunnable: Runnable = new Runnable {
-    private val removedExecutors = new mutable.HashSet[String]
+  private val executorRecoveryRunnable: Runnable = new Runnable {
+
+    private val MAX_EXECUTOR_LOST_REASON_CHECKS = 10
+    private val MAX_ALLOWED_EXECUTOR_RECOVERY_ATTEMPTS = 100
```

Comment:
I am dropping max allowed recovery attempts at it is not immediately obvious what should be a good default for this. Keeping this simple and similar to behavior on Yarn